core/injector/application_injector.cpp

```diff
@@ -4,3 +4,1305 @@
+    }
+    initialized = std::make_shared<network::RouterLibp2p>(
+        injector.template create<sptr<application::AppStateManager>>(),
+        injector.template create<libp2p::Host &>(),
+        injector.template create<const application::AppConfiguration &>(),
+        injector.template create<sptr<application::ChainSpec>>(),
+        injector.template create<network::OwnPeerInfo &>(),
+        injector.template create<sptr<network::StreamEngine>>(),
+        injector.template create<sptr<network::BabeObserver>>(),
+        injector.template create<sptr<consensus::grandpa::GrandpaObserver>>(),
+        injector.template create<sptr<network::SyncProtocolObserver>>(),
+        injector.template create<sptr<network::ExtrinsicObserver>>(),
+        injector.template create<sptr<network::Gossiper>>(),
+        injector.template create<const network::BootstrapNodes &>(),
+        injector.template create<sptr<blockchain::BlockStorage>>(),
+        injector.template create<sptr<libp2p::protocol::Ping>>(),
+        injector.template create<sptr<network::PeerManager>>(),
+        injector.template create<sptr<blockchain::BlockTree>>(),
+        injector.template create<sptr<crypto::Hasher>>());
+
+    return initialized.value();
+  }
+
+  template <class Injector>
+  sptr<network::PeerManager> get_peer_manager(const Injector &injector) {
+    static auto initialized =
+        boost::optional<sptr<network::PeerManager>>(boost::none);
+    if (initialized) {
+      return initialized.value();
+    }
+
+    initialized = std::make_shared<network::PeerManagerImpl>(
+        injector.template create<sptr<application::AppStateManager>>(),
+        injector.template create<libp2p::Host &>(),
+        injector.template create<sptr<libp2p::protocol::Identify>>(),
+        injector.template create<sptr<libp2p::protocol::kademlia::Kademlia>>(),
+        injector.template create<sptr<libp2p::protocol::Scheduler>>(),
+        injector.template create<sptr<network::StreamEngine>>(),
+        injector.template create<const application::AppConfiguration &>(),
+        injector.template create<const application::ChainSpec &>(),
+        injector.template create<sptr<clock::SteadyClock>>(),
+        injector.template create<const network::BootstrapNodes &>(),
+        injector.template create<const network::OwnPeerInfo &>(),
+        injector.template create<sptr<network::SyncClientsSet>>(),
+        injector.template create<sptr<blockchain::BlockTree>>(),
+        injector.template create<sptr<crypto::Hasher>>(),
+        injector.template create<sptr<blockchain::BlockStorage>>(),
+        injector.template create<sptr<network::BabeObserver>>());
+    return initialized.value();
+  }
+
+  template <typename Injector>
+  sptr<consensus::BlockExecutor> get_block_executor(const Injector &injector) {
+    static auto initialized =
+        boost::optional<sptr<consensus::BlockExecutor>>(boost::none);
+    if (initialized) {
+      return *initialized;
```

Comment:
For same-way-using:
```suggestion
      return initialized.value();
```
Please, check others